dev/src/index.ts

```diff
@@ -597,6 +597,23 @@ export class Firestore implements firestore.Firestore {
           // also set the `protocol` option for GAX fallback to force http
           if (useFallback) {
             settings.protocol = 'http';
+
+            // If the emulator mode is enabled, we set emulator credentials
+            // to prevent the gRPC-fallback client from trying to obtain
+            // service account credentials from the environment.
+            if (process.env.FIRESTORE_EMULATOR_HOST) {
+              const emulatorCredentials = {
+                access_token: 'owner',
+              };
+              const emulatorClient =
```

Comment:
Root cause for failing tests. The emulatorClient (cached credentials) needs to specify the project ID. If not specified here and not specified in the firestore settings, then the project ID will not be available, causing request the Firestore instance to behave incorrectly.

Reply:
I see! How does the system tests obtain the project id? We can use the firestore settings and `GCLOUD_PROJECT` env var here, if that makes sense.

```js
 emulatorClient.projectId = settings.projectId ?? process.env.GCLOUD_PROJECT;
```
WDYT?